src/core.ts

```diff
@@ -230,6 +250,10 @@ function isExternalizable(fnDef: FunctionDefinition): boolean {
     && !fnDef.returnParameters.parameters.some(p => p.typeName?.nodeType === 'Mapping' || p.typeName?.nodeType === 'FunctionTypeName');
 }
 
+function isNonPayableWithReturns(fnDef: FunctionDefinition): boolean {
```

Comment:
I think this should be `isNonViewWithReturns` right?

Reply:
yes